src/main/java/frc/robot/subsystems/SubsystemManager.java

```diff
@@ -8,22 +8,44 @@
 
 public class SubsystemManager {
     private DriveSubsystem driveSubsystem;
-
+    private NavXSubSystem navXSubSystem;
+    private VisionSubsystem visionSubsystem;
+    private ArmSubsystem armSubsystem;
+    
     public SubsystemManager() {
     }
 
     public DriveSubsystem getDriveSubsystem() {
         return driveSubsystem;
     }
 
+    public NavXSubSystem getNavXSubSystem() {
+        return navXSubSystem;
+    }
+    public VisionSubsystem getVisionSubsystem() {
+        return visionSubsystem;
+    }
 
+    public ArmSubsystem getArmSubsystem(){
+        return armSubsystem;
+    }
+    
     public void initAll() {
-        driveSubsystem = new DriveSubsystem();
-
+        navXSubSystem  = new NavXSubSystem();
+        driveSubsystem = new DriveSubsystem(getNavXSubSystem());
+        visionSubsystem = new VisionSubsystem();
+        armSubsystem = new ArmSubsystem();
+        
         // Initialize the known subsystems
+        navXSubSystem.initialize();
         driveSubsystem.initialize();
-
+        visionSubsystem.initialize();
+        armSubsystem.initialize();
+        
         // Make sure they all send their logging data
+        SmartDashboard.putData(navXSubSystem);
```

Comment:
i feel like there's a smart way to code these to avoid the repetition,by storing these in a list